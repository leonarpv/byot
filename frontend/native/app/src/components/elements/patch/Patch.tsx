import * as React from 'react';
import {StyleSheet, TouchableOpacityProps} from 'react-native';
import {Colors, TouchableOpacity} from 'react-native-ui-lib';

export type PatchProps = TouchableOpacityProps & {
  children?: React.ReactNode;
};

type Props = PatchProps;

interface State {}

const makeStyles = (props: Props, state: State) =>
  StyleSheet.create({
    root: {
      backgroundColor: Colors.g10,
      borderRadius: 8,
    },
  });

export const Patch: React.FC<Props> = (props: Props) => {
  const styles = makeStyles(props, {});
  return (
    <TouchableOpacity {...props} style={[props.style, styles.root]}>
      {props.children}
    </TouchableOpacity>
  );
};
