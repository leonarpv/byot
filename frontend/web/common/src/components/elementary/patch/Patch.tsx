import * as React from 'react';
import {makeStyles, Theme, Box, BoxProps} from '@material-ui/core';
import {WithStyles} from '../../../types/WithStyles';
import classNames from 'classnames';

interface Props extends WithStyles<typeof styles>, BoxProps {}

const styles = (theme: Theme) => ({
  root: {
    borderRadius: theme.shape.borderRadius,
    backgroundColor: theme.palette.background.paper,
  },
});
const useStyles = makeStyles(styles);

export const Patch: React.FC<Props> = (props: Props) => {
  const styles = useStyles(props);
  return <Box p={2} {...props} className={classNames(props.className, styles.root)} />;
};
