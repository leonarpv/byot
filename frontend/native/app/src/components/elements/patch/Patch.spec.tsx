import * as React from 'react';
import {render} from '@testing-library/react-native';
import {Patch} from './Patch';

describe('<Patch/>', () => {
  it('should render', () => {
    const {asJSON} = render(<Patch />);
    expect(asJSON()).toMatchSnapshot();
  });
});
