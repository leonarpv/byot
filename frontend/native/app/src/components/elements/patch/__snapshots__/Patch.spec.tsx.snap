// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Patch/> should render 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableOpacity
    style={
      Array [
        undefined,
        Object {
          "backgroundColor": "#FAFAF1",
          "borderRadius": 8,
        },
      ]
    }
  />
</View>
`;
